### PR TITLE
Create self-signed certificate KeyPair for mTLS

### PR DESCRIPTION
**Pull request by Alevsk**
- There's no need for creating a mTLS keypair based on a k8s CSR
- Will make Operator work on EKS (https://github.com/aws/containers-roadmap/issues/1604) by generating a self-signed certificate directly that will be trusted by MinIO and KES for mTLS authentication

# How to test

- Follow the steps on /src/github.com/Alevsk/minio-operator/docs/examples.md for `MinIO Tenant with Encryption enabled using Vault KMS`
- Once the tenant is deployed `port-forward` into the tenant
-  Validate encryption is configured correctly via `mc`, ie:

```
mc admin kms key status miniok8s --insecure
Key: my-minio-key
   - Encryption ✔
   - Decryption ✔
```

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

File: docs/tls.md [hunk 1]
```diff
@@ -76,10 +80,34 @@ spec:
...
+
+## Rotate MinIO Operator TLS certificates
+
+In case the operator TLS certificate expired you can just delete the `operator-tls` secret in the `minio-operator` namespace and
+operator will generate a new one automatically, additionally it will copy the `public.crt` portion of the certificate into
+each tenant namespace
+
+---
+
+## Using your own certificate for MinIO Operator
```
Comment (ravindk89) [inline docs/tls.md:101]: @Alevsk assuming once I do this, the same behavior aboe applies where operator auto-copies the public.crt over to each tenant

File: docs/tls.md [hunk 2]
```diff
@@ -76,10 +80,34 @@ spec:
...
+If you wish to use your own certificate for `MinIO Operator` instead of using the auto generated one, just delete or replace the content of the
+`operator-tls` secret in the `minio-operator` namespace, `operator-tls` must contain the `public.crt` and `private.key` keys.
+
+---
+
+## Using your own CA Certificate for MinIO Operator
+
+If your `MinIO` tenants are using `custom certificates` or certificates generated by your own internal `certificate authority` (ie: `cert-manager`).
+`MinIO Operator` needs to trust the `TLS` connections in order to talk to the `MinIO tenants`, for that you need to create a new secret in the
+`minio-operator` namespace named `operator-ca-tls`, inside this secret create a new key `ca.crt` that will include the public certificate
```
Comment (ravindk89) [inline docs/tls.md:112]: Can this have multiple ca.crt if for some reason users have multiple CA for tenants?

**Review by ravindk89 — DISMISSED** [review]
Minor questions, otherwise LGTM

**Comment by harshavardhana** [comment]
I do not understand the gist here what is this change for?

File: docs/tls.md (hunk 1 above)
Comment (Alevsk) [reply docs/tls.md:101]: @ravindk89 no, for that each tenant has the `externalCaCertSecret` in which the user can pass an array of secrets

**Comment by Alevsk** [comment]
> I do not understand the gist here what is this change for?

check this thread @harshavardhana https://github.com/aws/containers-roadmap/issues/1604 currently operator doesnt work properly on EKS when tenants are deployed with KES enabled because EKS will refuse to generate the mtls client certificates (at the moment on EKS you can only generate certificates with `server auth usage` but not `client auth usage` via kubernetes CSR).

**Comment by Alevsk** [comment]
@harshavardhana the docs say:

`Permitted key usages: Must not include usages beyond ["key encipherment", "digital signature", "server auth"]`

we need the `client auth` so EKS will not issue the required mTLS certificates for MinIO/KES

**Review by dvaldivia — APPROVED** [review]

File: docs/tls.md (hunk 1 above)
Comment (ravindk89) [reply docs/tls.md:101]: OK - so if the Operator rolls with custom CA, the user is responsible for passing that CA to `externalCACertSecret`.

We should include that  requirement in this section.

**Review by ravindk89 — APPROVED** [review]
LGTM mod one nit, just for the docs section.

**Pull request merged**